File: src/main/java/me/nepnep/nepbot/message/command/CommandRegister.java
package me.nepnep.nepbot.message.command;

import me.nepnep.nepbot.message.command.commands.*;
import me.nepnep.nepbot.message.command.commands.admin.*;
import me.nepnep.nepbot.message.command.commands.animals.*;
import me.nepnep.nepbot.message.command.commands.fun.*;
import me.nepnep.nepbot.message.command.commands.info.*;
import me.nepnep.nepbot.message.command.commands.mod.Ban;
import me.nepnep.nepbot.message.command.commands.mod.Kick;
import me.nepnep.nepbot.message.command.commands.mod.Mute;
import me.nepnep.nepbot.message.command.commands.mod.Purge;
import me.nepnep.nepbot.message.command.commands.mod.Unmute;

import java.util.HashMap;
import java.util.Map;

public class CommandRegister {
    public static final Map<String, ICommand> REGISTER = new HashMap<>();

    public static void registerCommands() {
        register(new Ping());
        register(new Help());
        register(new TinyRat());
        register(new BigRat());
        register(new Stat());
        register(new Kick());
        register(new Ban());
        register(new Mute());
        register(new Purge());
        register(new SetDefaultRole());
        register(new GiveDefaultRole());
        register(new Unmute());
        register(new Steal());
        register(new BigSnail());
        register(new TinySnail());
        register(new Http());
        register(new Xkcd());
        register(new BigDog());
        register(new XmasRat());
        register(new BigMonkey());
        register(new WhoIs());
        register(new Lmddgtfy());
        register(new BigEars());
        register(new LewdBlackListCommand());
        register(new RemoveLewdBlackList());
        register(new RandomChimpEvent());
        register(new Suggestion());
        register(new Shut());
        register(new ProfilePicture());
        register(new MinecraftServer());
        register(new BotInfo());
        register(new UrbanDictionary());
        register(new BottomCommand());
        register(new BottomWhiteListCommand());
        register(new RemoveBottomWhiteList());
        register(new JoinMessage());
        register(new Shutdown());
    }
    private static void register(ICommand command) {
        if (!REGISTER.containsKey(command.getInvoke())) {
            REGISTER.put(command.getInvoke().toLowerCase(), command);
        }
    }
}

File: src/main/kotlin/me/nepnep/nepbot/message/command/CommandRegister.kt
package me.nepnep.nepbot.message.command

import me.nepnep.nepbot.message.command.commands.*
import me.nepnep.nepbot.message.command.commands.`fun`.BottomCommand
import me.nepnep.nepbot.message.command.commands.`fun`.Lmddgtfy
import me.nepnep.nepbot.message.command.commands.`fun`.Xkcd
import me.nepnep.nepbot.message.command.commands.admin.*
import me.nepnep.nepbot.message.command.commands.animals.*
import me.nepnep.nepbot.message.command.commands.info.*
import me.nepnep.nepbot.message.command.commands.mod.*
import me.nepnep.nepbot.message.command.commands.music.*

object CommandRegister {
    val register = mutableMapOf<String, AbstractCommand>()

    fun registerCommands() {
        register(BottomWhitelist())
        register(GiveDefaultRole())
        register(JoinMessage())
        register(RemoveBottomWhitelist())
        register(SetDefaultRole())
        register(Steal())
        register(BigRat())
        register(HttpCat())
        register(XmasRat())
        register(BottomCommand())
        register(Lmddgtfy())
        register(Xkcd())
        register(BotInfo())
        register(MinecraftServer())
        register(ProfilePicture())
        register(UrbanDictionary())
        register(WhoIs())
        register(Ban())
        register(Kick())
        register(Mute())
        register(Purge())
        register(Unmute())
        register(Help())
        register(Ping())
        register(Shutdown())
        register(Suggestion())
        register(Poll())
        register(Prefix())
        register(Play())
        register(Skip())
        register(Stop())
        register(Search())
        register(Volume())
        register(StealSticker())
        register(Reload())
        register(Load())
        register(Unload())
        register(Info())
        register(SwitchPlayer())
    }

    fun register(command: AbstractCommand) {
        register[command.invoke] = command
    }
    
    fun remove(command: AbstractCommand) {
        register.remove(command.invoke)
    }
}
